src/sinks/amqp/tests.rs

```diff
@@ -0,0 +1,225 @@
+use super::*;
+
+#[test]
+pub fn generate_config() {
+    crate::test_util::test_generate_config::<AMQPSinkConfig>();
+}
+
+#[cfg(feature = "amqp-integration-tests")]
+mod integration_tests {
```

Comment:
The timing might not work out but I'm adding a `assert_sink_error` test helper in this PR:

https://github.com/vectordotdev/vector/pull/14225

In either case, is an error path test case achievable?